gui/common_qt_functions.py

```diff
@@ -174,40 +176,111 @@ def setComponentSettings(self):
             'hard-limit': False # not exposed
-        })
-        self.settings.setValue('attr_cache',{
+            }
+
+        allMountSettings['attr_cache'] = {
             'timeout-sec': 120,
             'no-cache-on-list': False,
             'enable-symlinks': False,
             # the following attr_cache settings are not exposed in the GUI
             'max-files': 5000000,
             'no-cache-dirs': False
-        })
-        self.settings.setValue('loopbackfs',{
+            }
+        
+        allMountSettings['loopbackfs'] = {
             'path': ''
-        })
+            }
 
-        self.settings.setValue('mountall',{
+        allMountSettings['mountall'] = {
             'container-allowlist': [],
             'container-denylist': []
-        })
-        self.settings.setValue('health_monitor',{
+            }
+        
+        allMountSettings['health_monitor'] = {
             'enable-monitoring': False,
             'stats-poll-interval-sec': 10,
             'process-monitor-interval-sec': 30,
             'output-path':'',
             'monitor-disable-list': []
-        })
-        self.settings.setValue('logging',{
+            }
+        
+        allMountSettings['logging'] = {
             'type' : 'syslog',
             'level' : 'log_err',
             'file-path' : '$HOME/.cloudfuse/cloudfuse.log',
             'max-file-size-mb' : 512,
             'file-count' : 10 ,
             'track-time' : False
-            })
+            }
+
+class customConfigFunctions():
+    def __init__(self):
+        super().__init__()
+
+    # defaultSettingsManager has set the settings to all default, now the code needs to pull in
+    #   all the changes from the config file the user provides. This may not include all the
+    #   settings defined in defaultSettingManager.
+    def initSettingsFromConfig(self):
+        dictForConfigs = self.getConfigs()
+        for option in dictForConfigs:
+            # check default settings to enforce YAML schema
+            invalidOption = not (option in self.settings)
+            invalidType = type(self.settings.get(option, None)) != type(dictForConfigs[option])
+            if invalidOption or invalidType:
+                print(f"WARNING: Ignoring invalid config option: {option} (type mismatch)")
+                continue
+            if type(dictForConfigs[option]) == dict:
+                tempDict = self.settings[option]
+                for suboption in dictForConfigs[option]:
+                    tempDict[suboption] = dictForConfigs[option][suboption]
+                self.settings[option]=tempDict
+            else:
+                self.settings[option] = dictForConfigs[option]
+
+    def getConfigs(self,useDefault=False):
+        workingDir = self.getWorkingDir()
+        if useDefault:
+            # Use programmed defaults
+            defaultSettingsManager.setAllDefaultSettings(self,self.settings)
+            configs = self.settings
```

Comment:
Since `getConfigs` is not part of the `defaultSettingsManager` class, `self` means an instance of `customConfigFunctions` here, so does this call to `setAllDefaultSettings` create a separate dictionary and write to the new `settings` field in `customConfigFunctions`?
Or is `self.settings` defined somewhere else in `customConfigFunctions`?

Reply:
Fixed